src/source.cpp

```diff
@@ -355,30 +344,57 @@ void free_memory_source()
   model::external_sources.clear();
 }
 
-// fill the source bank from the external source
-void fill_source_bank_custom_source()
+//Load custom source library
+void load_custom_source_library()
 {
 #ifdef HAS_DYNAMIC_LINKING
+
   // Open the library
-  auto source_library = dlopen(settings::path_source_library.c_str(), RTLD_LAZY);
-  if (!source_library) {
+  model::source_library = dlopen(settings::path_source_library.c_str(), RTLD_LAZY);
+  if (!model::source_library) {
     fatal_error("Couldn't open source library " + settings::path_source_library);
   }
 
   // reset errors
   dlerror();
 
   // get the function from the library
-  using sample_t = Particle::Bank (*)(uint64_t* seed);
-  auto sample_source = reinterpret_cast<sample_t>(dlsym(source_library, "sample_source"));
+  //using sample_t = Particle::Bank (*)(uint64_t* seed);
+  model::sample_source = reinterpret_cast<model::sample_t>(dlsym(model::source_library, "sample_source"));
```

Comment:
Since `sample_t` and the global variables in this file are not used externally, I would suggest using an anonymous namespace:
```C++
namespace {
  ...
  sample_t sample_source
  void* source_library
}
```
This would allow you to use the names within functions without calling out their namespace
```suggestion
  sample_source = reinterpret_cast<sample_t>(dlsym(source_library, "sample_source"));
```

Reply:
Suggestions have been implemented